<launch>
  <arg name="ns" default="tracking_ros" />
  <arg name="model_type" default="vit_h" />
  <arg name="device" default="cuda:0" />
  <arg name="text_prompt"/>
  <arg name="num_slots" default="-1"/>
  
  <arg name="compressed" default="false"/>
  <arg name="input_image" default="/kinect_head/rgb/image_rect_color"/>
  <arg name="processed_image" default="/$(arg ns)/processed_image"/>
  <arg name="segmentation_image" default="/$(arg ns)/segmentation_image" />
  <arg name="segmentation_mask" default="/$(arg ns)/segmentation_mask" />

  <group ns="$(arg ns)">
    <!-- process image and repub -->
    <node if="$(arg compressed)" name="raw_image_transport" pkg="image_transport" type="republish" args="compressed">
      <remap from="in" to="$(arg input_image)"/>
      <remap from="out" to="$(arg processed_image)"/>
    </node>
    <node unless="$(arg compressed)" name="raw_image_transport" pkg="image_transport" type="republish" args="raw">
      <remap from="in" to="$(arg input_image)"/>
      <remap from="out" to="$(arg processed_image)"/>
    </node>

    <!-- tracking node -->
    <include file="$(find tracking_ros)/launch/track_all_node.launch" >
      <arg name="model_type" value="$(arg model_type)" />
      <arg name="device" value="$(arg device)" />
      <arg name="num_slots" value="$(arg num_slots)" />
      <arg name="processed_image" value="$(arg processed_image)"/>
      <arg name="segmentation_image" value="$(arg segmentation_image)" />
      <arg name="segmentation_mask" value="$(arg segmentation_mask)" />
    </include>

  </group>
</launch>
